### PR TITLE
[#456] #EXTEND 'assemblyName: DotNet.Testcontainers; function: WithDo…

### PR DESCRIPTION
**Pull request by MarcinSzyszka**
…ckerEndpoint ' {Extended method with Docker.DotNet.Credentials parameter and made required changes to related code to pass credentials to Docker.DotNet.DockerClientConfiguration ctor }

### CODE REVIEW

**Review by HofmeisterAn — COMMENTED** [review]
Thanks for your contribution. I can do the review at the beginning of next month. Right now, I have limit access to a proper device — sorry for the delay. I think this relates to #370 too.

File: src/DotNet.Testcontainers/Builders/AbstractBuilder.cs [hunk 1]
```diff
@@ -1,7 +1,8 @@
-﻿namespace DotNet.Testcontainers.Builders
+namespace DotNet.Testcontainers.Builders
 {
   using System;
   using System.Collections.Generic;
+  using Docker.DotNet;
```
Comment (HofmeisterAn) [inline src/DotNet.Testcontainers/Builders/AbstractBuilder.cs:5]: The library (builder) does not expose models from this namespace. Please see the discussions in #370 too. I think it makes sense to combine the endpoint and credential in a single interface and offer different implementations specific for the authentication.

File: src/DotNet.Testcontainers/Builders/AbstractBuilder.cs [hunk 2]
```diff
@@ -34,6 +35,12 @@ public TBuilderEntity WithDockerEndpoint(string endpoint)
       return this.MergeNewConfiguration(new DockerResourceConfiguration(endpoint: new Uri(endpoint)));
     }
 
+    /// <inheritdoc cref="IAbstractBuilder{TBuilderEntity}" />
+    public TBuilderEntity WithDockerEndpoint(string endpoint, Credentials credentials)
+    {
+      return this.MergeNewConfiguration(new DockerResourceConfiguration(endpoint: new Uri(endpoint), credentials: credentials));
```
Comment (HofmeisterAn) [inline src/DotNet.Testcontainers/Builders/AbstractBuilder.cs:41]: We should use our own interface. The implementation should consider the common [env variables](https://docs.docker.com/compose/reference/envvars/#docker_host) (`DOCKER_HOST`, `DOCKER_TLS_VERIFY`, `DOCKER_CERT_PATH`) too (at least in the future). Then we can set and apply the default credentials from the constructor based on the machine configuration.

**Review by HofmeisterAn — COMMENTED** [review]
Please cover your changes with unit tests. I think we can exposes a secure Docker daemon on the Linux agent.

File: src/DotNet.Testcontainers/Builders/AbstractBuilder.cs (hunk 2 above)
Comment (MarcinSzyszka) [reply src/DotNet.Testcontainers/Builders/AbstractBuilder.cs:41]: If I implement our own interface and class for certificate credentials will it be acceptable for now?
I know that perfect soultion would be connecting to dockerd secured with tls automatically as it is done in java lib but it sounds like a lot of work and as we can see in https://github.com/HofmeisterAn/dotnet-testcontainers/issues/370 last comment was in the end of 2k21 ;)
I'd like to use this lib in my projects but without credentials support is useless now.
We could treat this solution as first step to handle connection to secured dockerd - I don't exclude my further contribution on that

Comment (HofmeisterAn) [reply src/DotNet.Testcontainers/Builders/AbstractBuilder.cs:41]: > We could treat this solution as first step to handle connection to secured dockerd

I can publish your changes in its own package as soon as I’m back. For develop, I would prefer the mentioned things above.

Comment (MarcinSzyszka) [reply src/DotNet.Testcontainers/Builders/AbstractBuilder.cs:41]: Ok, if I want to try to implement solution with docker env variables, should I do that in this PR or create new one?

Comment (HofmeisterAn) [reply src/DotNet.Testcontainers/Builders/AbstractBuilder.cs:41]: Yes, a new pull request would be nice. Smaller increments would be appreciated too (first the interface). I'll publish your package later that day.

Comment (HofmeisterAn) [reply src/DotNet.Testcontainers/Builders/AbstractBuilder.cs:41]: Your changes are published in `1.6.0-beta.GitHub-457`.

**Comment by HofmeisterAn** [comment]
I'll close this in favor of https://github.com/HofmeisterAn/dotnet-testcontainers/issues/370#issuecomment-1146842098. You can now use the latest version to configure / authenticate the client.

**Pull request closed without merging**